client/pages/saved-item-details.jsx

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import EmailForm from '../components/email-form-modal';
+import Redirect from '../components/redirect';
+import NotFound from './not-found';
+
+export default class SavedItemDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      listing: null,
```

Comment:
Try to come up with a way to use your ItemDetails component. What's keeping you from using that so that a whole new component is necessary? I'm honestly asking. Is it possible to re-use the other component?